mattermost/mattermost
# [GH-24981] Fix: Manually selecting EST timezones is confused with daylight savings

#### Summary

Selecting certain timezones led to multiple items being highlighted in the timezone dropdown. 

This PR ensures only one item is highlighted after the user selects a timezone.

#### Ticket Link


Fixes https://github.com/mattermost/mattermost/issues/24981
Jira https://mattermost.atlassian.net/browse/MM-50728

#### Screenshots


<details><summary>Before</summary>

https://github.com/user-attachments/assets/6416c92b-9fea-4944-8716-d09e59f97e6c

</details> 

<details><summary>After</summary>

https://github.com/user-attachments/assets/9b600042-23bf-4b89-848d-a77828a67c3f

</details> 

#### QA Review Steps

1. Click the Settings icon at the top right.
2. Click "Display" in the modal's sidebar then click "TImezone" in the main panel.
3. Uncheck "Automatic" if it's checked.
4. Select `UTC (-05:00) Eastern Time (US Canada)` from the dropdown.
5. Open the dropdown again and verify `UTC (-05:00) Eastern Time (US Canada)` is the only item highlighted.
6. Repeat steps 4 and 5 with `(UTC-07:00) Pacific Daylight Time (US & Canada)` and `(UTC) Edinburgh`.

#### Release Note

```release-note
Fixed bug where multiple timezones are highlighted when selecting certain timezones.
```
